src/main/java/Restriction.java

```diff
@@ -0,0 +1,24 @@
+import java.util.Arrays;
+
+public class Restriction {
+    private static final int CORRECT_INPUT_LENGTH = 3;
+    private static final int CORRECT_INPUT_CHAR_MAX = 57;
+    private static final int CORRECT_INPUT_CHAR_MIN = 49;
+
+    public boolean checkInputIsCorrect(String input) {
+        int inputLength = (int) Arrays.stream(input.split("")).distinct().count();
+        if (inputLength != CORRECT_INPUT_LENGTH) {
+            return false;
+        }
+        for (char number : input.toCharArray()) {
+            if (number < CORRECT_INPUT_CHAR_MIN || number > CORRECT_INPUT_CHAR_MAX) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public boolean checkSelectIsCorrect(String input) {
+        return input.equals(Choice.NEW_GAME_START) || input.equals(Choice.GAME_END);
```

Comment:
이 부분의 경우 input으로 ``null` 이 들어온다면 어떻게 될까요?  
아마도 **NullPointException**이 발생할 것입니다. 
```suggestion
        return Choice.NEW_GAME_START.equals(input) ||Choice.GAME_END.equals(input);
```
이 코드를 위와같이 변경하게 된다면, input이 null이더라도 NullPointException을 방지할 수있습니다. 
아래의 참고자료를 보시면서 그 이유를 알고 넘어가면 좋을 것 같습니다. 

[참고자료](https://zzikjh.tistory.com/entry/java-equal-%EC%A0%9C%EB%8C%80%EB%A1%9C-%EC%82%AC%EC%9A%A9%ED%95%98%EA%B8%B0)

Reply:
좋은 가르침 감사합니다. 공부하고 유념하도록 하겠습니다.